Backend/Remora.Discord.Caching/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -48,17 +50,20 @@ public static class ServiceCollectionExtensions
     /// <remarks>
     /// The cache uses a run-of-the-mill <see cref="IMemoryCache"/>. Cache entry options for any cached type can be
     /// configured using <see cref="IOptions{CacheSettings}"/>.
+    ///
+    /// When choosing a cache implementation, it should be noted that <see cref="AddDiscordCaching"/> and
```

Comment:
Is it not more appropriate to always require `AddDiscordCaching`, and have `AddDiscordRedisCaching` as an "addon" to the base caching services?

Reply:
The comment is moreso in reference to the behavior exhibited by Remora's caching; if you were to use Redis caching, Remora will always cache to Redis, and vice versa.

Reply:
Might want to reword this, then.

 Another (somewhat unrelated) thought - having both MemoryCacheProvider and RedisCacheProvider available (regardless of whether you've called AddDiscordRedisCaching) in the service provider might be useful, in the case where you want to cache data strictly in-process.

I'm thinking that instead of calling `Replace`, just adding the redis provider as another instance of `ICacheProvider` (thereby making that one take precedence) might be more appropriate; alternatively, adding both first as their plain types, and then making a separate forwarding call to `AddSingleton<ICacheProvider, ...>`.

Reply:
Doesn't this cause registration issues when the service provider attempts to resolve it? Or does it simply chose the last added type?

Reply:
If you register more than one service as a provider of the same interface, the last added type is used if the interface is requested. All services that implement that interface can be requested by using `GetServices`, or the individual services by their concrete types (if they've been registered as their concrete types).